server/py/services/api/initial_data.py

```diff
@@ -115,6 +116,14 @@ def init_data(
             logger.warning("Migrations failed, changing API state", state=state)
             config.httpdb.state = state
             raise
+
+    db_session = create_session()
```

Comment:
What about moving this to before the 'is_migration_needed', so we cam reuse the same session for both operations?